src/esp/nav/PathFinder.cpp

```diff
@@ -946,9 +953,46 @@ vec3f PathFinder::Impl::getRandomNavigablePoint(const int maxTries /*= 10*/) {
     ESP_ERROR() << "Failed to getRandomNavigablePoint.  Try increasing max "
                    "tries if the navmesh is fine but just hard to sample from";
     return vec3f::Constant(Mn::Constants::nan());
-  } else {
-    return pt;
   }
+  return pt;
+}
+
+vec3f PathFinder::Impl::getRandomNavigablePointAroundSphere(
+    const vec3f& circleCenter,
+    const float radius,
+    const int maxTries) {
+  if (getNavigableArea() <= 0.0)
+    throw std::runtime_error(
+        "NavMesh has no navigable area, this indicates an issue with the "
+        "NavMesh");
+
+  vec3f pt = vec3f::Constant(Mn::Constants::nan());
+  dtPolyRef start_ref = 0;  // ID to start our search
+  dtStatus status = navQuery_->findNearestPoly(
```

Comment:
This does have some edge cases with spheres that span multiple floors, but I am not sure the best way to resolve those. (It looks like it does a random selection from a search of all points discovered that fit into the circle from the starting ref).